src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -0,0 +1,299 @@
+            {
+                // Handles one directional character differences between roads. Meant to capture
+                // differences in directionality; e.g. in Pie St. N vs. Pie St. S, neither should be
+                // flagged as being inconsistent with one another.
+                if (!possibleDirectionalDifference && incomingEdgeNameIndex == startingEdgeNameIndex
+                        && incomingEdgeNameIndex < incomingEdgeName.length()
+                        && startingEdgeNameIndex < startingEdgeName.length())
+                {
+                    possibleDirectionalDifference = hasDirectionalCharacterDifference(
+                            incomingEdgeName.charAt(incomingEdgeNameIndex),
+                            startingEdgeName.charAt(startingEdgeNameIndex));
+                }
+
+                // Handles one number character differences between roads. Records if the
+                // startingEdgeName and incomingEdgeName differ by one number in their name strings.
+                if (!singleNumericalDifference && incomingEdgeNameIndex == startingEdgeNameIndex
+                        && incomingEdgeNameIndex < incomingEdgeName.length()
+                        && startingEdgeNameIndex < startingEdgeName.length())
+                {
+                    singleNumericalDifference = hasNumericalCharacterDifference(
+                            incomingEdgeName.charAt(incomingEdgeNameIndex),
+                            startingEdgeName.charAt(startingEdgeNameIndex));
+                }
+
+                if (incomingEdgeNameIndex == 0)
+                {
+                    results[incomingEdgeNameIndex][startingEdgeNameIndex] = startingEdgeNameIndex;
+                }
+
+                else if (startingEdgeNameIndex == 0)
+                {
+                    results[incomingEdgeNameIndex][startingEdgeNameIndex] = incomingEdgeNameIndex;
+                }
+
+                else
+                {
+                    results[incomingEdgeNameIndex][startingEdgeNameIndex] = min(
+                            results[incomingEdgeNameIndex - 1][startingEdgeNameIndex - 1]
+                                    + costOfSubstitution(
+                                            incomingEdgeName.charAt(incomingEdgeNameIndex - 1),
+                                            startingEdgeName.charAt(startingEdgeNameIndex - 1)),
+                            results[incomingEdgeNameIndex - 1][startingEdgeNameIndex] + 1,
+                            results[incomingEdgeNameIndex][startingEdgeNameIndex - 1] + 1);
+                }
+            }
+        }
+        // If there's only a single character difference and that character is a directional OR
+        // numerical character, we consider both roads to be different and so we don't flag them.
+        // Else we return the Levenshtein distance as usual.
+        return (possibleDirectionalDifference || singleNumericalDifference)
```

Comment:
One way to make this more general for other users would be to return the actual edit operations necessary to get from one string to another. In our case, we can search those operations for a directional difference or a numerical difference and update as appropriate. In other cases, a user could take the raw levenshtein distance by counting the number of operations, etc. [Take a look at this stackoverflow post for an example of what I'm talking about, along with sample implementations](https://stackoverflow.com/questions/10638597/minimum-edit-distance-reconstruction/10639697#10639697).

If you took that step, I think this would be very useful outside of this class.

Reply:
Thanks. I've decided to take a more general approach to numerical/directional differences by wrapping them in what I refer to as Identifier Strings. These Identifiers are meant to handle road names whose numbers/directions differ by more than a single character.

I do think we should track all edit operations for our utility, though. I can see it being useful if we wanted to dig into patterns of TPs vs FPs.